pkg/destroy/aws/aws.go

```diff
@@ -301,9 +310,10 @@ func (o *ClusterUninstaller) findEC2Instances(ctx context.Context, ec2Client *ec
 								instanceLogger.Info("Terminated")
 								deleted.Insert(arn)
 							}
-						} else {
-							resources = append(resources, arn)
+						} else if *instance.State.Name != "shutting-down" {
+							resourcesRunning = append(resourcesRunning, arn)
 						}
+						resourcesNotTerminated = append(resourcesNotTerminated, arn)
```

Comment:
You are adding all instances to `resourcesNotTerminated`, even instances that are already terminated.
```
if terminated
  add to delete
  continue for loop
if not shutting down
  add to resources running
add to resources not terminated
```
or
```
switch state
case terminated
  add to delete
case shutting down
  add to resources running
  fallthrough
default
  add to resources not terminated
```

Reply:
This is still adding terminated instances to `resourcesNotTerminated`.